src/System.Net.WebSockets.Client/tests/ClientWebSocketTest.cs

```diff
@@ -372,8 +412,73 @@ public static IEnumerable<object[]> UnavailableWebSocketServers
                 var recvSegment = new ArraySegment<byte>(receiveSegment.Array, receiveSegment.Offset, recvRet.Count);
                 Assert.Equal(message, WebSocketData.GetTextFromBuffer(recvSegment));
             }
-        }        
-#endregion
+        }
+
+        [ConditionalTheory(nameof(WebSocketsSupported)), MemberData(nameof(EchoServers))]
+        public async Task SendReceive_VaryingLengthBuffers_Success(Uri server)
+        {
+            using (ClientWebSocket cws = await WebSocketHelper.GetConnectedWebSocket(server, TimeOutMilliseconds, _output))
+            {
+                var rand = new Random();
+                var ctsDefault = new CancellationTokenSource(TimeOutMilliseconds);
+
+                // Values chosen close to boundaries in websockets message length handling.
+                foreach (int bufferSize in new int[] { 1, 125, 126, 127, 128, ushort.MaxValue - 1, ushort.MaxValue, ushort.MaxValue + 1, ushort.MaxValue * 2 }) 
```

Comment:
We have tests elsewhere for 0.  I don't think it's particularly interesting for what this test is trying to handle.